### PR TITLE
server : tuning tests

### PR DESCRIPTION
**Pull request by ggerganov**
- Fix incorrect passing of the temperature as the string "0.8f"
- Increase timeout to make the thread sanitizer workflow pass
- Pass temperature explicitly where it matters

### CODE REVIEW

File: examples/server/tests/features/steps/steps.py [hunk 1]
```diff
@@ -883,7 +883,7 @@ async def request_completion(prompt,
                                     "cache_prompt": cache_prompt,
                                     "id_slot": id_slot,
                                     "seed": seed if seed is not None else 42,
-                                    "temperature": temperature if temperature is not None else "0.8f",
+                                    "temperature": temperature if temperature is not None else 0.8f,
```
Comment (JohannesGaessler) [inline examples/server/tests/features/steps/steps.py:886]:
```suggestion
                                    "temperature": str(temperature) if temperature is not None else 0.8f,
```

In order to always pass the same data type.

**Review by JohannesGaessler — APPROVED** [review]
Ignore my previous comment, I had mixed up old and new.

File: examples/server/tests/features/results.feature [hunk 2]
```diff
@@ -13,6 +13,7 @@ Feature: Results
 
   Scenario Outline: consistent results with same seed
     Given <n_slots> slots
+    And   0.0 temperature
```
Comment (JohannesGaessler) [inline examples/server/tests/features/results.feature:16]: This should be set to temperature 1, not temperature 0. With temperature 0 you get greedy decoding in which case the seed is irrelevant. So this test can no longer detect issues with e.g. the RNG state being shared between slots.

Comment (ggerganov) [reply examples/server/tests/features/results.feature:16]: Ah yes, will push a fix now

**Pull request merged**